myFitnessTrainer/src/components/GenerateNewPlanAlgoButton.js

```diff
@@ -0,0 +1,130 @@
+import { collection, getDocs, query, where, and } from "firebase/firestore";
+import { db } from "../../firebaseConfig";
+import { auth } from "../../firebaseConfig";
+import { getDocumentId, postDocument } from "../../databaseFunctions";
+
+/*
+    Pass in the following props as the below types:
+    - goal: Array<String>
+    - level: String
+    - duration: Int
+    - modifications: Array<String>
+*/
+const GenerateNewPlanAlgoButton = ({ goal, level, duration }) => {
+    // Test props for debugging
+    goal = goal ?? "strength";
+    level = level ?? "beginner";
+    duration = duration ?? 5;
+    // modifications = modifications ?? ["legs"];
+
+    const navigation = useNavigation();
+
+    /*
+    performs query to db for exercises collection based on goal and level survey result and returns an array of exercises
+    */
+    const getExercises = async (goal, level) => {
+        const q = query(
+            collection(db, "exercises"),
+            filterExerciseQuery(goal, level)
+        );
+        const querySnapshot = await getDocs(q);
+        const exercises = [];
+        querySnapshot.forEach((doc) => {
+            exercises.push(doc.data());
+        });
+        return exercises;
+    };
+
+    const createPlan = (exercises, duration) => {
+        const workoutPlan = [];
+
+        for (let i = 0; i < exercises.length && i < duration; i++) {
+            let randomIdx = Math.floor(Math.random() * exercises.length);
+            workoutPlan.push(exercises[randomIdx]);
+        }
+        return workoutPlan;
+    };
+
+    // TODO: Need to add ability to filter with modifications
+    const filterExerciseQuery = (goal, level) => {
+        switch (level) {
+            case "beginner":
+                return and(
+                    where("fitness_goal", "==", goal),
+                    where("fitness_level", "==", "beginner")
+                );
+            case "intermediate":
+                return and(
+                    where("fitness_goal", "==", goal),
+                    where("fitness_level", "==", "intermediate")
+                );
+            case "advanced":
+                return and(
+                    where("fitness_goal", "==", goal),
+                    where("fitness_level", "==", "advanced")
+                );
+            default:
+                break;
+        }
+    };
+    /* 
+        1. query all exercises with criteria goal, level, and modifications
+        2. create plan by selecting number of exercises == duration, random based from query 
+        3. add this plan to workoutplans collection
+    */
+    const handleCreateNewPlanAlgo = async () => {
+        const retrievedExercises = await getExercises(goal, level);
+        const workoutPlan = createPlan(retrievedExercises, duration);
+        const userId = await getDocumentId(
+            "users",
+            "email",
+            auth.currentUser.email
+        );
+
+        const postObject = {
```

Comment:
As I am thinking about figuring out how to mark their workout plan as 'complete' after they have completed 5 days (like our test data duration), I think we should also add a new field to our WorkoutPlans called 'days_completed'. We can increment this each time they complete their workout plan for the day. We would then check that to see if they completed the workout plan..and also can report on it in the workout summary. Is that something you think we should add?

Also, there was a field we had planned on called 'active' which is set to true to begin with. Then set to false when they have completed the plan. I still think that field is worth adding. should just be adding a line to this postObejct with `active: true`.

I like having the start_date as a timestamp so it can just be a number and we can convert when we need to access it.

Reply:
Thanks for catching the missing field. I added it in.